### PR TITLE
DOCS-8805 proxy config

### PR DESCRIPTION
**Pull request by sathya0**

### CODE REVIEW

**Review by andrei-m-sv — DISMISSED** [review]

File: modules/ROOT/pages/HTTP-based-connectors.adoc [hunk 1]
```diff
@@ -105,11 +105,91 @@ The HttpService allows you to obtain the HttpClient associated with a separate <
...
++
+[source,java,linenums]
+----
+.setProxyConfig(connectionParams.getProxyConfig())
+----
++
+
++
+For more information, refer to the https://github.com/mulesoft/mule-http-connector/blob/9914ce6aa6ceb3aa16a5b821289f15b9c6af6acc/src/main/java/org/mule/extension/http/api/request/proxy/DefaultProxyConfig.java[Java code]
+of how HTTP Connector handles the proxy.
```
Comment (hannanelson) [inline modules/ROOT/pages/HTTP-based-connectors.adoc]:
```suggestion
for how HTTP Connector handles the proxy.
```

File: modules/ROOT/pages/HTTP-based-connectors.adoc [hunk 2]
```diff
@@ -105,11 +105,91 @@ The HttpService allows you to obtain the HttpClient associated with a separate <
...
+            </oauth2:token-request>
+            <spring:property name="proxyConfig" ref="proxyConfig"/>
+        </oauth2:authorization-code-grant-type>
+</http:request-config>
+----
++
+
++
+Note that there must be a `@Parameter` annotation in the connection provider with
+the proxyConfig type so the connection refers to the proxy, for example:
```
Comment (hannanelson) [inline modules/ROOT/pages/HTTP-based-connectors.adoc]: When using the `proxyConfig` type, the connection provider must include a `@Parameter` annotation so that the connection refers to the proxy, for example:

Comment (hannanelson) [reply modules/ROOT/pages/HTTP-based-connectors.adoc]: Trying to remove the passive voice (and the "note that" construction), but not sure if that suggestion is technically correct. The sentence is a little hard to parse.

File: modules/ROOT/pages/HTTP-based-connectors.adoc [hunk 3]
```diff
@@ -105,11 +105,91 @@ The HttpService allows you to obtain the HttpClient associated with a separate <
...
+----
+@Parameter
+@Optional
+@Placement(tab = "Proxy", order = 1)
+private MyConnectorProxyConfiguration proxyConfig;
+----
++
+
++
+Also, note that the proxy configuration must be set in the HTTP Client:
```
Comment (hannanelson) [inline modules/ROOT/pages/HTTP-based-connectors.adoc]:
```suggestion
You must also set the proxy configuration in the HTTP Client:
```

**Review by hannanelson — DISMISSED** [review]

**Review by hannanelson — DISMISSED** [review]

**Review by hannanelson — APPROVED** [review]
